packages/components/src/pages/VsCodeExtension.vue

```diff
@@ -72,40 +98,160 @@
           />
         </template>
         <template v-slot:controls>
-          <v-popper-menu :showDot="filtersChanged">
+          <v-popper-menu :isHighlight="filtersChanged">
             <template v-slot:icon>
               <FilterIcon class="control-button__icon" />
             </template>
             <template v-slot:body>
-              <h2>Filters</h2>
-              <div>
-                <input
-                  type="checkbox"
-                  id="limit-root-events"
-                  v-model="filters.limitRootEvents.on"
-                  @change="setUserFiltered"
-                />
-                <label for="limit-root-events">Limit root events to HTTP</label>
-              </div>
-              <div>
-                <input
-                  type="checkbox"
-                  id="unlabeled-events"
-                  v-model="filters.unlabeledEvents.on"
-                  @change="setUserFiltered"
-                />
-                <label for="unlabeled-events">Show unlabeled events</label>
-              </div>
-              <div>
-                <input
-                  type="checkbox"
-                  id="hide-media-requests"
-                  v-model="filters.hideMediaRequests.on"
-                  @change="setUserFiltered"
-                />
-                <label for="hide-media-requests"
-                  >Hide media HTTP requests</label
-                >
+              <div class="filters">
+                <div class="filters__head">
+                  <FilterIcon class="filters__head-icon" />
+                  <h2 class="filters__head-text">Filters</h2>
+                  <button class="filters__head-reset" @click="resetFilters()">
+                    Reset all
+                    <ResetIcon />
+                  </button>
+                </div>
+                <div class="filters__block">
+                  <div class="filters__block-head">
+                    <h3 class="filters__block-title">Root</h3>
+                    <form class="filters__form">
+                      <input
+                        class="filters__form-input"
+                        type="text"
+                        placeholder="add new root..."
+                      />
+                      <PlusIcon class="filters__form-plus" />
+                    </form>
+                  </div>
+                  <div class="filters__block-body filters__block-body--flex">
+                    <div
+                      class="filters__root"
+                      v-for="co in filters.rootObjects"
+                      :key="co.id"
```

Comment:
The filters applied won't always map to a `CodeObject` `id`. For example, `package:app/*`. So there needs to be a separate list of filters from which the `rootObjects` array is populated.